editor/store/selectors.js

```diff
@@ -300,6 +300,40 @@ export function isEditedPostEmpty( state ) {
 	);
 }
 
+/**
+ * Returns true if the post can be autosaved, or false otherwise.
+ *
+ * @param  {Object}  state Global application state
+ * @return {boolean}       Whether the post can be autosaved
+ */
+export function isPostAutosaveable( state ) {
+	// If the post is autosaving, it is not autosaveable.
+	if ( state.isAutosaving ) {
+		return false;
+	}
+
+	// If we don't already have an autosave, the post is autosaveable.
+	if ( ! hasAutosave( state ) ) {
+		return true;
+	}
+
+	const title = getEditedPostAttribute( state, 'title' );
+	const excerpt = getEditedPostExcerpt( state );
+	const content = getEditedPostContent( state );
+	const autosave = state.autosave;
+
+	// If the title, excerpt or content has changed, the post is autosaveable.
+	return (
+		( title !== autosave.title.raw ) ||
+		( excerpt !== autosave.excerpt.raw ) ||
+		( content !== autosave.content.raw )
+	);
+}
+
+export function hasAutosave( state ) {
```

Comment:
Can we add a JSDoc.

Reply:
Unit tests would be good too.

Reply:
👍

Reply:
@aduth added tests in https://github.com/WordPress/gutenberg/pull/6257/commits/c31eb055d276ea4fe44bbbd1764eecaaa3fad920